src/rbd.cc

```diff
@@ -3034,6 +3064,28 @@ if (!set_conf_param(v, p1, p2, p3)) { \
   // the relevant parts
   set_pool_image_name(imgname, (char **)&poolname,
 		      (char **)&imgname, (char **)&snapname);
+
+  if(!is_valid_name(poolname) || !is_valid_name(snapname) || !is_valid_name(imgname)) {
```

Comment:
Make sense.
Previous changes of mine were calling is_valid_name() for the same name
multiple times, which is not needed. Did necessary changes as suggested by
you.

On Tue, Mar 31, 2015 at 2:14 AM, David Zafman <notifications@github.com>
wrote:

> In src/rbd.cc
> <https://github.com/ceph/ceph/pull/4172#discussion_r27429934>:
>
> > @@ -3034,6 +3064,28 @@ if (!set_conf_param(v, p1, p2, p3)) { \
> >    // the relevant parts
> >    set_pool_image_name(imgname, (char **)&poolname,
> >  		      (char **)&imgname, (char **)&snapname);
> > +
> > +  if(!is_valid_name(poolname) || !is_valid_name(snapname) || !is_valid_name(imgname)) {
>
> All these if statements should be a lot simpler. Build an error string
> from the names "pool" "snap" "image" and depending on how many invalid
> items either separate with a "," and/or "and."
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ceph/ceph/pull/4172/files#r27429934>.
>